### PR TITLE
Al/optimize multiplication memory usage

### PR DESCRIPTION
**Pull request by agnesLeroy**
closes: _please link all relevant issues_

### Check-list:

* [ ] Tests for the changes have been added (for bug fixes / features)
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Relevant issues are marked as resolved/closed, related issues are linked in the description
* [ ] Check for breaking changes (including serialization changes) and add them to commit message following the conventional commit [specification][conventional-breaking]

[conventional-breaking]: https://www.conventionalcommits.org/en/v1.0.0/#commit-message-with-description-and-breaking-change-footer

### CODE REVIEW

**Comment by agnesLeroy** [comment]
Bench ran locally on p3.8xlarge: 64 bit multiplication goes down to 221ms (instead of 232ms on main)

**Comment by bbarbakadze** [comment]
good to go unless the test failure is not related to this PR.

File: backends/tfhe-cuda-backend/cuda/src/integer/multiplication.cuh [hunk 1]
```diff
@@ -248,6 +249,40 @@ __host__ void host_integer_sum_ciphertexts_vec_kb(
...
+  if (reused_lut == nullptr) {
+    release_reused_lut = true;
+    size_t ch_amount = r / chunk_size;
+    if (!ch_amount)
+      ch_amount++;
+    reused_lut = new int_radix_lut<Torus>(streams, gpu_indexes, gpu_count,
+                                          mem_ptr->params, 2,
+                                          2 * ch_amount * num_blocks, true);
+  }
+  int_radix_lut<Torus> *luts_message_carry = reused_lut;
```
Comment (bbarbakadze) [inline backends/tfhe-cuda-backend/cuda/src/integer/multiplication.cuh:265]: can we somehow check here that reused_lut has big enough pbs_buffer?

Comment (agnesLeroy) [reply backends/tfhe-cuda-backend/cuda/src/integer/multiplication.cuh]: I've made a change to do this, now we only reuse the lut if it meets requirements.

Comment (agnesLeroy) [reply backends/tfhe-cuda-backend/cuda/src/integer/multiplication.cuh]: Hmm doesn't look so easy, let me roll back the change.

**Review by bbarbakadze — APPROVED** [review]

**Pull request merged**